src/halmos/cheatcodes.py

```diff
@@ -866,8 +866,8 @@ def handle(sevm, ex, arg: ByteVec, stack) -> ByteVec | None:
 
         # vm.sign(uint256 privateKey, bytes32 digest) returns (uint8 v, bytes32 r, bytes32 s)
         elif funsig == hevm_cheat_code.sign_sig:
-            key = extract_bytes(arg, 4, 32)
-            digest = extract_bytes(arg, 4 + 32, 32)
+            key = uint256(extract_bytes(arg, 4, 32)).as_z3()
+            digest = uint256(extract_bytes(arg, 4 + 32, 32)).as_z3()
```

Comment:
q: any better way to do the same thing?

Reply:
It is readable but a bit too general. This will:
- grab the slice as a ByteVec
- unwrap() the slice, returning the underlying bytes or z3 object after calling simplify()
- wrap it in a BV (will call simplify() on it again if it is a z3 object)
- then finally wrap the underlying value in a z3 object if it is an int or return it

A more efficient way to do this would be something like:

```py
word = extract_word(arg, 4)
key = word if is_bv(word) else con(word)
```

or as a one-liner:

```py
key = (word := extract_word(arg, 4)) if is_bv(word) else con(word)
```

Reply:
yes i was afraid of that, and curious if there's a helper function for this. i will go back to using bytes_to_bv_value().

Reply:
https://github.com/a16z/halmos/pull/507/commits/32c978a51a6866a2cb358975cb4ff6d9431783db